### DISCUSSION TITLE
How to clear the ztoc digest if we remove soci index?

### DISCUSSION THREAD

**Discussion by katturaja-entropik** [awslabs/soci-snapshotter#1197] [Q&A]
I am able to clear the social index using
```
sudo soci index rm {{index-sha-value}}
```
But this does not clear the ztoc - Table of contents, Still I can see the deleted soci index ztoc using the below command
```
sudo soci ztoc ls
```
**Output:**
DIGEST                                                                     SIZE        LAYER DIGEST                                                               
sha256:132b8e18c8e358604007acdfd11a8b59849adee418ec5eb3d18ce032fb67dd02    3147000     

If we are not cleared in the system, it can cause system slowdown and memory bloat.
I would appreciate it if we could receive valuable answers.

**Comment by austinvazquez** [comment]
Hi @katturaja-entropik, thanks for using SOCI.

Can you try ```sudo soci rebuild-db```? We have to do this after an index remove when using the containerd content store to clear the database of the zTOCs which have been deleted.

**Reply by katturaja-entropik** [reply]
@austinvazquez Facing an issue 
soci: failed to add new artifacts to db: content digest sha256:3a5723e0382e9344b4ec75fa2151557598f464d5b1caacd66ea5a43d103b50c2: not found

**Reply by katturaja-entropik** [reply]
@austinvazquez 
<img width="1120" alt="image" src="https://github.com/awslabs/soci-snapshotter/assets/105864680/1eed04ac-492f-400c-b517-5667d362af03">
